src/middleware/logger/index.ts

```diff
@@ -83,7 +82,8 @@ export const logger = (fn: PrintFunc = console.log): MiddlewareHandler => {
   return async function logger(c, next) {
     const { method } = c.req
 
-    const path = getPath(c.req.raw)
+    const url = new URL(c.req.raw.url)
+    const path = url.pathname + url.search
```

Comment:
If a large size of query parameters are intentionally sent by an attacker, I believe this should be optional as it will contaminate the standard output.

Reply:
The same is true even if path is simply long.
So there is room for optimization here, but I don't think you need to worry about URL length.

Reply:
Yes, it is true.  
It might be nice to have the following features.

`/too/longggggggg~/paths` => `/too/longgggggg...`
`/?too=longggggggg~&pathssssssssss~` => `/?too=longg...`

Reply:
This is optimized on https://github.com/honojs/hono/pull/3702/commits/5a8553ba75068148c97b9906a9ca806896fa31c1 for performance.

> It might be nice to have the following features.
> 
> /too/longggggggg~/paths => /too/longgggggg...
> /?too=longggggggg~&pathssssssssss~ => /?too=longg...

It's an issue that potentially already exists. So that seems outside the scope of this PR.
You may want to create a new PR. Thanks.

Reply:
okay, thanks